client/blocks/login/social.jsx

```diff
@@ -182,12 +190,21 @@ class SocialLoginForm extends Component {
 						responseHandler={ this.handleGoogleResponse }
 						uxMode={ uxMode }
 						redirectUri={ this.getRedirectUrl( uxMode, 'google' ) }
-						onClick={ this.trackLogin.bind( null, 'google' ) }
+						onClick={ this.trackLoginAndRememberRedirect.bind( null, 'google' ) }
+						socialServiceResponse={
+							this.props.socialService === 'google' ? this.props.socialServiceResponse : null
```

Comment:
As a suggestion for a future improvement, I would consider rearchitecting this mechanism. From what I'm reading here, it appears to me that this component can receive both Google and Apple responses as a prop, and needs to look at which one it's getting in order to populate the correct subcomponent's props?

That seems to lead to a lot of special casing everywhere, so perhaps it would make sense to extract this logic somehow. Perhaps add an Apple-specific and Google-specific layer in between this and the buttons? Sorry I can't provide a more detailed alternative, but I'm still trying to grok the whole thing 😕

Reply:
Agreed it's not the cleanest thing and could benefit a refactor 👍